widgets/src/main/java/org/pentaho/mantle/client/dialogs/folderchooser/FolderTree.java

```diff
@@ -160,7 +160,7 @@ private String buildFetchTreeModelUrl( @Nullable String expandedPath ) {
       + "&showHidden=" + showHiddenFiles;
 
     if ( expandedPath != null ) {
-      url += "&expandedPath=" + URL.encodeQueryString( expandedPath );
+      url += "&expandedPath=" + URL.encodeQueryString( GenericFileNameUtils.encodePath( expandedPath ) );
```

Comment:
Did you face any issues with special characters in the expanded path parameter?
AFAIK, the only problem was in the below `buildFetchSubtreeModelUrl` call, for when a generic path is part of the URL path.
Generic paths in URL query string parameters should not need the special "generic file URL _path_" encoding, which exists mostly to get rid of the `/` characters which would interfere with the URL path structure, but not with the structure of a query string parameter.

Reply:
You are correct, we just need to do URL encoding here.